src/libstore/derivations.hh

```diff
@@ -15,13 +16,14 @@ namespace nix {
 struct DerivationOutput
 {
     StorePath path;
-    std::string hashAlgo; /* hash used for expected hash computation */
-    std::string hash; /* expected hash, may be null */
-    DerivationOutput(StorePath && path, std::string && hashAlgo, std::string && hash)
+    std::optional<FileSystemHash> hash; /* hash used for expected hash computation */
```

Comment:
Does this make sense here?

Reply:
That is, a derivation might not have any file ingestion at all. If it's not fixed output, I think it will only have file ingestion through its inputs.

Reply:
I would recommend just using the `Hash` type here.

Reply:
Very confusingly, the `hashAlgo` string also contained the `r:`. See https://github.com/NixOS/nix/pull/3439, where this comes from.